packages/vue/src/composables/useAuth.ts

```diff
@@ -1,43 +1,29 @@
 import { useActor } from '@xstate/vue';
-import { ref, reactive, Ref, watchEffect } from 'vue';
-import { getServiceFacade } from '@aws-amplify/ui';
+import { ref, reactive, Ref } from 'vue';
+import { AuthenticatorServiceFacade } from '@aws-amplify/ui';
 import { facade } from './useUtils';
 import { InterpretService } from '@/components';
 
 const service = ref() as Ref<InterpretService>;
-const useAuthenticatorValue = reactive({
-  ...facade,
-  send: '' as unknown,
-  state: '' as unknown,
-}) as any;
 
+/** @deprecated For internal use only */
 export const useAuth = (serv?: InterpretService) => {
   if (serv) {
     service.value = serv;
   }
   return useActor(service.value);
 };
 
-const useInternalAuthenticator = () => {
-  createValues();
-  watchEffect(() => {
-    createValues();
-  });
-  return useAuthenticatorValue;
-};
-
-export const useAuthenticator = useInternalAuthenticator;
-
-function createValues() {
-  if (!service.value) return;
+const useAuthInternal = reactive<AuthenticatorServiceFacade>({
+  ...facade,
+});
 
-  const { state, send } = useAuth();
+export const useAuthenticator = () => useAuthInternal;
 
-  const facadeValues = getServiceFacade({ send, state: state.value });
+export function updateValues(facadeValues: AuthenticatorServiceFacade) {
   for (const key of Object.keys(facade)) {
     //@ts-ignore
-    useAuthenticatorValue[key] = facadeValues[key];
+    useAuthInternal[key as keyof typeof useAuthInternal] =
```

Comment:
I'm close in removing this ts-ignore, but still having issues with null.

Reply:
Obviously not certain on this, but this reads to me as an upstream type issue with `facade`

Reply:
That's the unusual thing, facade is of type `AuthenticatorServiceFacade` and so is `useAuthInternal` so the keys exactly match. It's something to do with null being a type in the `useAuthInternal[key]`.  I'm up for suggestions. Otherwise maybe I can push this fix to a future PR.

Reply:
Think we should probably dig in deeper to understand this better, concerned with the casting